org-cyf-sdc/content/tools/sprints/3/process_spelunking_workshop.md

```diff
@@ -0,0 +1,539 @@
+ariane     66393  0.0  0.0  10072  1604 pts/0    R+   20:17   0:00 ps aux
+```
+(You'll see a lot more than I'm showing here.)
+<!--
+  I left out many from this output, so it wouldn't be so long.
+  I hope that won't confuse our students.
+  If I didn't do that, the output would be 123 lines, most of them kernel threads.
+  -->
+
+In the example output, you can see the program on the second-to-last line:
+```
+ariane     66392  0.0  0.2  17328  8784 pts/1    S+   20:17   0:00 python3 exercise_1.py
+```
+
+You can also see that the first column shows the username.
+So now you know my name. :)
+{{</tabs>}}
+
+In the `ps` output, you can see multiple columns.
+The one we're interested in, is the *PID* column.
+*PID* is short for *Process Identifier*.
+(*ID* is short for Identifier.)
+
+It turns out, working with text is a lot of work for computers.
+They much prefer working with numbers.
+And that's why the kernel doesn't think about a process as a name, but as a number: the *PID*.
+
+By default, `ps` (without any options) will only show (some of) our own processes, that have a
+{{<tooltip text="TTY" title="TTY">}}
+*TTY* is short for *TeleTYpe*, it's a kind of really old type-writer that used a phone line, so that what you type appears in some far-away place.
+{{</tooltip>}}.
+A *TTY* is used whenever you log in, and is the console used by programs.
+If you've ever heard of `standard in` or `standard out`, those are usually writing to a *TTY*, which is used by a person.
+
+The *STAT* and *TIME* columns tell us what a process is doing, and how much time it has used.
+We won't care about them today.
+
+{{<tabs>}}
+===[[Exercise]]===
+There are much more processes, but many don't have a *TTY*.  
+Can you figure out what the *TTY* of the `cron` process is?
+
+===[[Answer]]===
+The `cron` program is on a line similar like this:
+```
+USER         PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
+root       63213  0.0  0.0   6896  3108 ?        Ss   Feb22   0:00 /usr/sbin/cron -f -P
+```
+
+You can see its *TTY* column is `?`.
+This means the process doesn't have a *TTY*, and doesn't directly interact with a human.
+Many system services, and also programs which only interact with a person using graphics, won't have a *TTY*.
+{{</tabs>}}
+
+> [!TIP]
+>
+> Personally, I usually type `ps aux | grep exercise_1` when I want to find the *PID* of the exercise.
+> So if you only want to remember one thing, remember this. :)
+
+If you want to learn more, you can use `man ps` to read the documentation.
+It'll also have examples, and explain what the `a`, `u`, and `x` mean.
+
+{{<note title="Clean up">}}
+Now that you know how to find a *Process Identifier*, we no longer need the program that we started on the first terminal.
+You can hit "enter" to get it to stop.
+{{</note>}}
+
+### What You've Learned
+
+You've learned how you can use `ps` to find the *PID* of a process.
+You've learned the difference between a program and a process.
```

Comment:
Markdown rendering doesn't insert newlines where you're wrapping text over lines like this - are you expecting it to? If so, you can do one of:
1. Add a trailing `\` at the end of the line
2. Add an extra `\n` between lines you want separate
3. Add a manual `<br />` tag

(I'm not sure whether you're expecting newlines here, but have noticed you're doing manual wrapping at some sentence boundaries so figured I'd mention :))

Reply:
thank you, done.
(In all 3 What You've Learned sections.)